42201.md

```diff
@@ -0,0 +1,89 @@
+
+# ZenIP-42201
+
+    ZenIP: 42201
+    Title: Sidechain Versions
+    Owner: Paolo Tagliaferri, <paolotagliaferri@horizenlabs.io>; cronic, <cronic@horizenlabs.io>
+    Discussions-To: Paolo Tagliaferri, <paolotagliaferri@horizenlabs.io>
+    Comments-URI: https://horizen.global/invite/discord
+    Status: Draft 
+    Type: Consensus
+    Created: 2022-04-07
+    License: MIT
+
+## Table of Contents
+
+<!--ts-->
+
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Specification](#specification)
+
+
+## Abstract
+
+This ZenIP describes the consensus rule changes of the proposed enhancements to Zen v3.1.0. The consensus rule changes will activate via a hard fork at mainchain block height #1127000.
+The consensus rules will include the following changes:
+* a new field `version` is added to sidechain creation transactions (txtype -4) inside the Sidechain Creation Output (vsc_ccout)
+* the treasury addresses for ZBF Treasury, Secure Nodes and Super Nodes will change (coinbase transaction vout[1,2,3])
+
+In addition the following API changes are made:
+* new fields added to `getblocktemplate` RPC command
+* new API to retrieve all versions for specified sidechain IDs
+* new value pair added to extend the `getscgenesisinfo` RPC command
+
+
+## Motivation
+
+The enhancements fix an issue with the management of `Custom Fields` included in certificates. They also extend functionality for block template requests. These enhancements have already been implemented with v3.1.0.
+
+The change of Treasury addresses is motivated by the changing global security environment. To increase security, the 3-of-5 multisig signature scheme is changed to 4-of-7.
+
+## Specification
+
+**Sidechain Versioning**
+
+In previous zend versions, the `withdrawalEpoch` field was a 4 byte unsigned integer containing the withdrawal epoch length. The maximum valid withdrawal epoch length was limited or capped to 4032.
+In zend 3.1.0, the `withdrawalEpochLength` field is split into two separate fields, as follows.
+1. A 1 byte field for `version`
+2. A 3 byte field for `withdrawalEpochLength`
+             
+
```

Comment:
Insert the figure from https://github.com/HorizenOfficial/zen/pull/435#issue-1128409734

Reply:
adding image